### PR TITLE
feat: openlineage listener captures hook-level lineage

### PR DESCRIPTION
**Pull request by mobuchowski**
Enable OpenLineage provider to get data collected using [AIP-62 mechanism](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-62+Getting+Lineage+from+Hook+Instrumentation)

### CODE REVIEW

File: airflow/providers/openlineage/extractors/manager.py [hunk 1]
```diff
@@ -113,6 +115,10 @@ def extract_metadata(self, dagrun, task, complete: bool = False, task_instance=N
                 self.log.warning(
                     "Failed to extract metadata using found extractor %s - %s %s", extractor, e, task_info
                 )
+        elif IS_AIRFLOW_2_10_OR_HIGHER and (hook_lineage := self.get_hook_lineage()) is not None:
+            inputs, outputs = hook_lineage
+            task_metadata = OperatorLineage(inputs=inputs, outputs=outputs)
+            return task_metadata
```
Comment (kacpermuda) [inline airflow/providers/openlineage/extractors/manager.py]: Shouldn't we also try to use hook-level lineage's inputs and outputs when there is an extractor but inputs/outputs are missing, before we move to inlets and outlets there?

File: airflow/providers/openlineage/extractors/manager.py [hunk 2]
```diff
@@ -105,14 +106,25 @@ def extract_metadata(self, dagrun, task, complete: bool = False, task_instance=N
                 task_metadata = self.validate_task_metadata(task_metadata)
                 if task_metadata:
                     if (not task_metadata.inputs) and (not task_metadata.outputs):
-                        self.extract_inlets_and_outlets(task_metadata, task.inlets, task.outlets)
-
+                        if (
+                            IS_AIRFLOW_2_10_OR_HIGHER
```
Comment (kacpermuda) [inline airflow/providers/openlineage/extractors/manager.py]: nit: we can probably remove `IS_AIRFLOW_2_10_OR_HIGHER` check from this whole func as the ImportError in `self.get_hook_lineage()` will be raised anyway. But if you think it's worth keeping, we can also move it inside of `self.get_hook_lineage()` and simplify this if.
```suggestion
                        if (hook_lineage := self.get_hook_lineage()) is not None:
```

**Review by kacpermuda — APPROVED** [review]
LGTM

File: airflow/providers/openlineage/extractors/manager.py (hunk 2 above)
Comment (JDarDagran) [reply airflow/providers/openlineage/extractors/manager.py]: agreed with Kacper 👍

Comment (mobuchowski) [reply airflow/providers/openlineage/extractors/manager.py]: Fixed.

File: airflow/io/path.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def __getattr__(self, name):
         if callable(attr):
             # If the attribute is a method, wrap it in another method to intercept the call
             def wrapper(*args, **kwargs):
-                self.log.error("Calling method: %s", name)
+                self.log.debug("Calling method: %s", name)
```
Comment (eladkal) [inline airflow/io/path.py]: Is this related to this PR?
If not please extract to separated PR

Comment (mobuchowski) [reply airflow/io/path.py]: @eladkal done. Mistakenly attached change from https://github.com/apache/airflow/pull/41615

File: airflow/providers/openlineage/provider.yaml [hunk 4]
```diff
@@ -48,6 +48,7 @@ versions:
 dependencies:
   - apache-airflow>=2.8.0
   - apache-airflow-providers-common-sql>=1.6.0
+  - apache-airflow-providers-common-compat>=1.1.0
```
Comment (eladkal) [inline airflow/providers/openlineage/provider.yaml]:
```suggestion
  - apache-airflow-providers-common-compat>=1.2.0
```

Doesn't change much but just to have less versions to look on

**Review by eladkal — APPROVED** [review]
